### DISCUSSION TITLE
Invalid message Error in subscription

### DISCUSSION THREAD

**Discussion by amani-eric** [enisdenjo/graphql-ws#452] [Q&A] [answered]
**Screenshot**
![Screenshot 2023-02-27 at 21 23 29](https://user-images.githubusercontent.com/104340923/221791215-6eb14a91-63d5-4354-935f-10b31cc71715.png)

**Expected Behaviour**

It should not throw error and subscribe to data

**Actual Behaviour**

It throws `Invalid message error` without any more details on why the error is occuring and get disconnected

**Debug Information**

The subscription configuration seems fine and connection get successfull initially. 
It disconnect whenever this error occurs .

![Screenshot 2023-02-28 at 09 55 16](https://user-images.githubusercontent.com/104340923/221789124-63a992fc-082a-4b01-babe-e8d79932493d.png)

**Further Information**

I'm using NestJS v`9.0.0` graphql-ws v`5.11.3`

**Comment by enisdenjo** [comment, marked as answer]
Are you sure you're using graphql-ws on both sides (client and server)? [Note that graphql-ws is **not** backwards compatible with the legacy subscriptions-transport-ws protocol.](https://github.com/enisdenjo/graphql-ws#disclaimer)

Furthermore, there is no error with just the message `Invalid message` anywhere in graphql-ws codebase ([see here](https://github.com/search?q=repo%3Aenisdenjo%2Fgraphql-ws%20%22Invalid%20message%22&type=code)).
